mesh/mesh.cpp

```diff
@@ -5828,7 +5828,6 @@ void Mesh::SetNodes(const Vector &node_coord)
 void Mesh::NewNodes(GridFunction &nodes, bool make_owner)
 {
    if (own_nodes) { delete Nodes; }
-   nodes.Pull();
```

Comment:
Why is this no longer necessary?

Reply:
I have moved it to for now on the application's side, like in [Laghos](https://github.com/CEED/Laghos/blob/3d35ede905bd5f8259f3f500c00620ce255b2212/laghos_solver.cpp#L411).
I wanted to reduce the number of `Push`/`Pull`, to get a better understanding where they are really needed. We can re-introduce the ones that really make sense.